inc/class-statify-frontend.php

```diff
@@ -92,11 +92,23 @@ public static function track_visit( $is_snippet = false ) {
 		}
 
 		/* Relative target url */
-		$data['target'] = user_trailingslashit( str_replace( home_url( '/', 'relative' ), '/', $target ) );
+		$data['target'] = str_replace( home_url( '/', 'relative' ), '/', $target );
+		// Maybe add trailing slash if that is no search query.
+		if ( 0 !== strpos( $data['target'], '/?s=' ) ) {
+			$data['target'] = user_trailingslashit( $data['target'] );
+		}
 
 		// Trim target url.
 		if ( $wp_rewrite->permalink_structure ) {
-			$data['target'] = wp_parse_url( $data['target'], PHP_URL_PATH );
+			if ( 0 !== strpos( $data['target'], '/?s=' ) ) {
+				// No search, so we only need the `PHP_URL_PATH`.
+				$data['target'] = wp_parse_url( $data['target'], PHP_URL_PATH );
+			}
+		}
+
+		// Convert search query to lowercase.
+		if ( 0 === strpos( $data['target'], '/?s=' ) ) {
+			$data['target'] = strtolower( $data['target'] );
```

Comment:
We could think of slightly refactoring this routine to not repeat the search-check thrice in a row.

Something like this should do the same
```php
if ( 0 !== strpos( $data['target'], '/?s=' ) ) {
	// Add trailing slash if that is no search query.
	$data['target'] = user_trailingslashit( $data['target'] );

	// Trim target url.
	if ( $wp_rewrite->permalink_structure ) ) {
		// No search, so we only need the `PHP_URL_PATH`.
		$data['target'] = wp_parse_url( $data['target'], PHP_URL_PATH );
	}
} else {
	// Convert search query to lowercase.
	$data['target'] = strtolower( $data['target'] );
}
```

Reply:
Good idea, modified that